File: beecrowd/1043 - Triângulo.go
package main

import "fmt"

func isTriangle(a, b, c float64) bool {
	return a < b+c && b < a+c && c < a+b
}

func trianglePerimeter(a, b, c float64) float64 {
	return a + b + c
}

func trapeziumArea(a, b, c float64) float64 {
	return (a + b) * c / 2.0
}

func main() {

	var a, b, c float64

	fmt.Scan(&a, &b, &c)

	if isTriangle(a, b, c) {
		fmt.Printf("Perimetro = %.1f\n", trianglePerimeter(a, b, c))
	} else {
		fmt.Printf("Area = %.1f\n", trapeziumArea(a, b, c))
	}
}

File: beecrowd/1043 - Triângulo.kt
import java.util.*

class Console {
    private val buffer = ArrayDeque<String>()

    private fun fillBuffer() {
        if(buffer.isEmpty()) {
            val str = readLine()!!
            val pieces = str.split(" ")

            for (piece in pieces) {
                buffer.add(piece)
            }
        }
    }

    fun readInt(): Int {
        fillBuffer()

        val result = buffer.removeFirst()

        return result.toInt()
    }

    fun readDouble(): Double {
        fillBuffer()

        val result = buffer.removeFirst()

        return result.toDouble()
    }

    fun readString(): String {
        fillBuffer()

        val result = buffer.removeFirst()

        return result
    }

    fun readLong(): Long {
        fillBuffer()

        val result = buffer.removeFirst()

        return result.toLong()
    }

    fun readFloat(): Float {
        fillBuffer()

        val result = buffer.removeFirst()

        return result.toFloat()
    }
}

fun isTriangle(a: Double, b: Double, c: Double): Boolean {
    val sides = arrayOf(a, b, c)

    sides.sort()

    return sides[2] < sides[0] + sides[1]
}

fun perimeter(a: Double, b: Double, c: Double): Double {
    return a + b + c
}

fun trapezeArea(a: Double, b: Double, c: Double): Double {
    return (a + b) * c / 2
}

fun main(args: Array<String>) {

    val console = Console()

    val a = console.readDouble()
    val b = console.readDouble()
    val c = console.readDouble()

    if (isTriangle(a, b, c)) {
        println("Perimetro = %.1f".format(Locale.US, perimeter(a, b, c)))
    } else {
        println("Area = %.1f".format(Locale.US, trapezeArea(a, b, c)))
    }
}